File: src/main/java/rss/entity/Menu.java
package rss.entity;

public class Menu {
    private int id;
    private String nom;
    private Burger burger;
    private Complement boisson;
    private Complement frites;
    private double prixTotal;
    private String imageUrl;
    private String description;

    public Menu() {}

    public Menu(String nom, Burger burger, Complement boisson, Complement frites,
                String imageUrl, String description) {

        this.nom = nom;
        this.burger = burger;
        this.boisson = boisson;
        this.frites = frites;
        this.imageUrl = imageUrl;
        this.description = description;

        this.prixTotal = burger.getPrix() + boisson.getPrix() + frites.getPrix();
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public void setPrixTotal(double prixTotal) { this.prixTotal = prixTotal; }

    public String getNom() { return nom; }
    public Burger getBurger() { return burger; }
    public Complement getBoisson() { return boisson; }
    public Complement getFrites() { return frites; }
    public double getPrixTotal() { return prixTotal; }
    public String getImageUrl() { return imageUrl; }
    public String getDescription() { return description; }
}

File: Models/Menu.cs
namespace BrasilBurgerClient.Models
{
    using System.ComponentModel.DataAnnotations.Schema;
    [Table("menu")]
    public class Menu
    {
        public int Id { get; set; }
        public string Nom { get; set; }

        // 🔗 relations
        public int BurgerId { get; set; }
        public Burger Burger { get; set; }

        public int BoissonId { get; set; }
        public Complement Boisson { get; set; }

        public int FritesId { get; set; }
        public Complement Frites { get; set; }

        public double PrixTotal { get; set; }
        public string ImageUrl { get; set; }
        public string? Description { get; set; }
        public bool Archived { get; set; }
    }
}
